src/solc_0.7/catalyst/AssetAttributesRegistry.sol

```diff
@@ -171,10 +172,12 @@ contract AssetAttributesRegistry is WithMinter, WithUpgrader {
         uint64 blockNumber
     ) internal {
         require(msg.sender == _minter || msg.sender == _upgrader, "NOT_AUTHORIZED_MINTER");
+        // TODO: verify audit comment
         require(gemIds.length <= MAX_NUM_GEMS, "GEMS_MAX_REACHED");
```

Comment:
is line 176 redundant ? as we check the gemIds.length on line 178

Reply:
It is 2 different checks
the first one ensure you can never have a number of gems higher that the maximum possible number of gems.
the second check if the number of gems exceed the max number of gem for a particular catalyst.

the first could indeed be removed, but its gas cost is negligeable and the check avoid the risk of caused by adding a misconfigured catalyst that would return a max gems greater than MAX_NUM_GEMS